### DISCUSSION TITLE
Do I need Prisma to generate database types?

### DISCUSSION THREAD

**Discussion by jsjoeio** [supabase/supabase#2890] [Questions] [answered]
I was originally going to use Prisma to generate types based on my Supabase tables, but it sounds like the docs [here](https://supabase.io/docs/reference/javascript/generating-types) say I can do this with the third-party tool OpenAPI?

To put simply, 

_You don't need Prisma in your stack. You can generate database types in TypeScript by following the docs [here](https://supabase.io/docs/reference/javascript/generating-types)_.

^Confirming this is correct?

**Comment by soedirgo** [comment, marked as answer]
Hey @jsjoeio, yes you can use the OpenAPI tool, but it's not very sophisticated - we're working on our own type dumper (tracked here: https://github.com/supabase/postgres-meta/issues/103) but until then you might be better served with Prisma.

**Reply by jsjoeio** [reply]
Ah got it. I'll subscribe to that issue then. Thank you!
